internal/service/streaming_processor.go

```diff
@@ -0,0 +1,506 @@
+					allErrors = append(allErrors, *result.ErrorSummary)
+				}
+			}
+
+			chunk = nil // Reset chunk
+			chunkIndex++
+		}
+	}
+
+	// Process remaining records in the last chunk
+	if len(chunk) > 0 {
+		result, err := sp.processChunkWithRetry(ctx, processor, chunk, headers, chunkIndex, config)
+		if err != nil {
+			sp.Logger.Error("failed to process final chunk", "chunk_index", chunkIndex, "error", err)
+			allErrors = append(allErrors, fmt.Sprintf("Final chunk %d: %v", chunkIndex, err))
+		} else {
+			totalProcessed += result.ProcessedRecords
+			totalSuccessful += result.SuccessfulRecords
+			totalFailed += result.FailedRecords
+			if result.ErrorSummary != nil {
+				allErrors = append(allErrors, *result.ErrorSummary)
+			}
+		}
+	}
+
+	return sp.finalizeProcessing(t, totalProcessed, totalSuccessful, totalFailed, allErrors, chunkIndex)
+}
+
+// finalizeProcessing updates the task with final processing results
+func (sp *StreamingProcessor) finalizeProcessing(
+	t *task.Task,
+	totalProcessed int,
+	totalSuccessful int,
+	totalFailed int,
+	allErrors []string,
+	chunkIndex int,
+) error {
+	// Update final task status
+	t.ProcessedRecords = totalProcessed
+	t.SuccessfulRecords = totalSuccessful
+	t.FailedRecords = totalFailed
+
+	if len(allErrors) > 0 {
+		errorSummary := strings.Join(allErrors, "; ")
+		t.ErrorSummary = &errorSummary
+	}
+
+	sp.Logger.Infow("completed streaming processing",
+		"task_id", t.ID,
+		"total_processed", totalProcessed,
+		"successful", totalSuccessful,
+		"failed", totalFailed,
+		"chunks_processed", chunkIndex+1)
+
+	return nil
+}
+
+// processChunkWithRetry processes a chunk with retry logic using battle-tested backoff
+func (sp *StreamingProcessor) processChunkWithRetry(
+	ctx context.Context,
+	processor ChunkProcessor,
+	chunk [][]string,
+	headers []string,
+	chunkIndex int,
+	config *StreamingConfig,
+) (*ChunkResult, error) {
+	// Configure exponential backoff
+	backoffConfig := backoff.NewExponentialBackOff()
+	backoffConfig.MaxElapsedTime = 5 * time.Minute
+	backoffConfig.InitialInterval = config.RetryDelay
+	backoffConfig.MaxInterval = 30 * time.Second
+
+	var chunkResult *ChunkResult
+	operation := func() error {
+		var err error
+		chunkResult, err = processor.ProcessChunk(ctx, chunk, headers, chunkIndex)
+		return err
+	}
+
+	err := backoff.Retry(operation, backoffConfig)
+	if err != nil {
+		sp.Logger.Warnw("chunk processing failed after retries",
+			"chunk_index", chunkIndex,
+			"error", err)
+		return nil, ierr.WithError(err).
+			WithHint("Failed to process chunk after retries").
+			WithReportableDetails(map[string]interface{}{
+				"chunk_index": chunkIndex,
+				"max_retries": config.MaxRetries,
+			}).
+			Mark(ierr.ErrValidation)
+	}
+
+	return chunkResult, nil
+}
+
+// downloadFileStream downloads a file and returns a stream using retryable HTTP client
+func (sp *StreamingProcessor) downloadFileStream(ctx context.Context, t *task.Task) (io.ReadCloser, error) {
```

Comment:
One problem with keeping everything in memory during the whole processing is that it is going to take RAM usage
Ideally we should store the file on the disk
then partially read the disk stored file in the stream processor and at the end of processing - delete the file